### PR TITLE
fix: Adding overall format before appending ColumnFormatCodes

### PR DESCRIPTION
**Pull request by masterkidan**
Currently the encode method for copy_both_response is missing the OverallFormat as part of the 1st byte. This causes errors when using it. 

The issue is mainly seen while creating a 'mock' back-end for tests.

### CODE REVIEW

**Review by gitstashpop — CHANGES_REQUESTED** [review]
Hi, thanks for the patch!

I noticed that `copy_{in,out}_response` also have the same issue. Since these are all related, would you mind fixing that as well?

A unit test for at the least, encoding would also be awesome as well. Thanks!

**Comment by masterkidan** [comment]
@gitstashpop I see that copy_{[in](https://github.com/jackc/pgproto3/blob/40ecac487c46292c2139a1fc7a7365be2a287ba6/copy_in_response.go#L52), [out](https://github.com/jackc/pgproto3/blob/40ecac487c46292c2139a1fc7a7365be2a287ba6/copy_out_response.go#L51)}_response do encode the OverallFormat ... Can you clarify the issue that you are referencing here?

**Comment by gitstashpop** [comment]
> @gitstashpop I see that copy_{[in](https://github.com/jackc/pgproto3/blob/40ecac487c46292c2139a1fc7a7365be2a287ba6/copy_in_response.go#L52), [out](https://github.com/jackc/pgproto3/blob/40ecac487c46292c2139a1fc7a7365be2a287ba6/copy_out_response.go#L51)}_response do encode the OverallFormat ... Can you clarify the issue that you are referencing here?

Ah I missed that. Just the unit test then please.

**Comment by masterkidan** [comment]
@gitstashpop added.

File: copy_both_response_test.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		OverallFormat:     byte(1), // Just to differ from defaults
+		ColumnFormatCodes: []uint16{0, 1},
+	}
+	dstBytes := []byte{}
+	dstBytes = src.Encode(dstBytes)
+	dst := pgproto3.CopyBothResponse{}
+	err := dst.Decode(dstBytes[5:])
+	assert.NilError(t, err, "No errors on decode")
+	assert.Equal(t, dst.OverallFormat, src.OverallFormat, "OverallFormat is decoded successfully")
+	assert.DeepEqual(t, dst.ColumnFormatCodes, src.ColumnFormatCodes)
```
Comment (gitstashpop) [inline copy_both_response_test.go:21]: I think a meaningful test here would be to have a byte array with data you would expect on the wire, decode, encode, and then verify the encoded data equals what you initially saw.

**Review by gitstashpop — CHANGES_REQUESTED** [review]

File: go.sum [hunk 2]
```diff
@@ -9,6 +9,7 @@ github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZN
 github.com/stretchr/objx v0.1.0/go.mod h1:HFkY916IF+rwdDfMAkV7OtwuqBVzrE8GR6GFx+wExME=
 github.com/stretchr/testify v1.4.0 h1:2E4SXV/wtOkTonXsotYi4li6zVWxYlZuYNCXe9XRJyk=
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405 h1:yhCVgyC4o1eVCa2tZl7eS0r+SDo693bJlVdllGtEeKM=
```
Comment (gitstashpop) [inline go.sum:12]: is this just leftover from the other iteration? probably can be dropped

**Review by gitstashpop — APPROVED** [review]

**Comment by jackc** [comment]
LGTM

**Pull request merged**